devito/builtins.py

```diff
@@ -53,6 +54,131 @@ def smooth(f, g, axis=None):
         dv.Operator(dv.Eq(f, g.avg(dims=axis)), name='smoother')()
 
 
+def gaussian_smooth(f, sigma=1, _order=4, mode='reflect'):
+    """
+    Gaussian smooth function.
+    """
+    class DataDomain(dv.SubDomain):
```

Comment:
Is it worth moving this inside Devito?
BTW, I wouldn't call it "DataDomain" -- all subdomains are data domains after all ?

Reply:
> Is it worth moving this inside Devito?
Not at the moment I don't think - if we're going to have some 'pre-set' subdomain methods it's probably better we introduce them after a little more thought. Or is that not what you meant?

>BTW, I wouldn't call it "DataDomain" -- all subdomains are data domains after all ?
Yeah, sure. I think `ObjectiveDomain` is a better name.